new_01-vim-beginner.tutor

```diff
@@ -0,0 +1,1234 @@
+
+---> The brown fox jumps under the bus. {expect:The brown fox jumps over the lazy dog.}
+---> The brown fox jumps over the lazy dog. {expect:The brown fox jumps over the lazy dog.}
+
+It is not always a good idea to search for the character we want. If it's a very
+common character we will waste keystrokes skipping over the false positives.
+Often times it's better to jump to a nearby but less common character and then
+correct the position by one or two.
+
+# Lesson something.something: UNDO, REDO AND REPEAT
+
+** We can undo and redo changes. More imporantly though, we can repeat them **
+
+To undo a change press «u», press «C-R» to redo it again.
+
+  1. Make some change to the line to break it.
+
+  2. Undo your change by pressing «u».
+
+  3. You can re-apply the change pressing «C-R».
+
+---> Everything is great. {expect:Everything is great.}
+
+In the lessons above I have been metioning repeatability of actions and here is
+where it comes into play. We can repeat any last action using the period «.»
+operator.
+
+  1. Navigate to the first wrong word and change it with «ce».
+
+  2. After the change exit insert mode immediately and jump to the next word.
+
+  3. Now press «.» and see the change take effect immediately.
+
+---> Roses are blue, and tomatoes are blue as well. {expect:Roses are red, and tomatoes are red as well.}
+---> Roses are red, and tomatoes are red as well. {expect:Roses are red, and tomatoes are red as well.}
+
+An action consits of a count, an operator and a motion, you know that already.
+What I did not tell you is that everything you type in isert mode is part of
+the action as well. This means you can repeat entire phrases, sentences or even
+paragraphs. The only catch is that you cannot leave insert mode or it ends the
+action.
+
+The dot is actually a macro and always contains the last action. We will
+later learn how to define our own macros which are more powerful than that. We
+will also learn how to transplant text properly.
+
+The dot is still very powerful and you should use it whenever the possibility
+arises. Like most other operators it can be combined with a motion.
+
+## Lesson something.something: THE PUT COMMAND
+
+** Typ p to put previously deleted text after the cursor. **
+
+The «p» operator can put previously deleted text after the cursor, and the
+upper-case variant places it in front of the cursor.
+
+  1. Type  «dd»  on the d) line to delete the.
+
+  2. Move the cursor to the c) line, ABOVE where the deleted line should go.
+
+  3. Type   «p»   to put the line below the cursor.
+
+  4. Now delete the a) line below and move the cursor to the b) line.
+
+  5. Type   «P»   to put the line above the cursor.
+
+---> d) Can you learn too? {expect:NULL}
+---> b) Violets are blue, {expect:NULL}
+---> c) Intelligence is learned, {expect:NULL}
+---> a) Roses are red, {expect:NULL}
+
+The deleted text is stored in a register and always contains the last deletion
```

Comment:
Buffer is a no-go, as the term means something very specific in Vim. how about:

> The deleted text is stored in a so-called register, memory holding some text
temporarliy, and always contains the last deletion or yank (see next lesson).